.github/workflows/docker.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Docker images
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/docker.yml'
+      - 'docker/**'
+  schedule:
+    - cron: '42 7 * * *' # run at 7:42 UTC (morning) every day
+jobs:
+  build:
+    name: Build and push
+    runs-on: ubuntu-latest
+    env:
+      RUST_IMAGE_TAG: latest
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          docker login --username $DOCKERHUB_USERNAME --password $DOCKERHUB_TOKEN
+
+          RUST_IMAGE=rust:$RUST_IMAGE_TAG
+          docker run --rm $RUST_IMAGE rustc --version
```

Comment:
GitHub Action's `ubuntu-latest` has Rust built-in, so there's no need to use a container for it.

Reply:
This line is actually a leftover from my debugging - I've removed it. I've noticed it thanks to your comment.

The next line `RUST_VERSION=$(docker run ...` needs to use rustc from `$RUST_IMAGE`, because its version needs to match rustc from $CHEF_IMAGE (which is itself based on $RUST_IMAGE).